### PR TITLE
Translations update from Hosted Weblate

### PR DESCRIPTION
**Pull request by weblate**
Translations update from [Hosted Weblate](https://hosted.weblate.org) for [Nicotine+/nicotine-plus](https://hosted.weblate.org/projects/nicotine-plus/nicotine-plus/).

Current translation status:

Weblate translation status

### CODE REVIEW

**Comment by hboetes** [comment]
@SnIPeRSnIPeR made a typo, I think.

**Comment by slook** [comment]
No, I think `#` might be intentional, since this string is a column header that is for numeric Queue Position... This column displays very narrow data (only 1 or 2 characters), but his language has a very long string for this which would be inappropriate here.

**Comment by hboetes** [comment]
The translation for Queue position in Russian you can find in the diff, it's a few chars bigger. And after reading the header, I know what it says, and then I make the column smaller. When the column header is '#' I bet there are people who have no idea what it means will have to look it up, and they can't make the column a bit bigger.

I understand the idea, but I'm not in favour of it.

**Comment by slook** [comment]
Agree, since we don't have a tooltip or other means of knowing the columns meaning, the string should be reverted to the proper label.

**Comment by SnIPeRSnIPeR** [comment]
> No, I think `#` might be intentional, since this string is a column header that is for numeric Queue Position... This column displays very narrow data (only 1 or 2 characters), but this language has a very long string for this which would be inappropriate here.
> 
> We should consider changing the source string to something shorter like this, and perhaps put the long string in a tooltip.
> 
> I have my column width set so that only a single letter `Q` is visible, yet the data are plenty visible:

# is using in qBittorrent, № in uTorrent. Many other programs use # or № instead of other words or phrases like "Порядок очереди", because # or № is short. Yes, I chahged original phrase because it takes up a lot of space. Also, many people know what # or № means.

**Comment by SnIPeRSnIPeR** [comment]
> I understand the idea, but I'm not in favour of it.

What about "№"?

**Comment by hboetes** [comment]
If there is a term which has any sort of problem, first look at the originating string. If you think that one can be improved we have a win for everyone. So please bring it up and we can have a look at it together.

Now that I think about it, simply using the word "queue" is a lot shorter and also explains what the column does, and sufficiently small that it doesn't even need wrapping.

**Comment by hboetes** [comment]
Added comment for clarity:

https://hosted.weblate.org/translate/nicotine-plus/nicotine-plus/en/?q=Queue+Position&sort_by=-priority%2Cposition&offset=1#comments

**Comment by slook** [comment]
> many people know what "#" or "№" means.

`#` or `№` are ambiguous with a list "Index", ie the order in which items are added, rather than relating to the "Queue".

Your idea has merit, we should look into a shorter source string so that all languages can benefit. However, strings are frozen until after the release of 3.2.1 so this can't be done now, the string "Queue" doesn't exist at the moment. We might as well leave it the way it was for the time being, since this is a non-urgent issue.

**Comment by hboetes** [comment]
> Perhaps it would be useful to add more comments to other strings to help the translators identify what UI component is being modified, because context doesn't seem that easy to discover without looking at the code base.

Absolutely. Especially for those short strings.

**Comment by slook** [comment]
Simply commenting "Short string for small button" or "Short string for narrow column header" would suffice, although also stating exactly which part(s) of the UI the string appears would be ideal, especially for those areas where clarity/brevity is more important than proper grammar.

**Comment by SnIPeRSnIPeR** [comment]
> If there is a term which has any sort of problem, first look at the originating string. If you think that one can be improved, we have a win for everyone. So please bring it up, and we can have a look at it together.
> 
> Now that I think about it, simply using the word "queue" is a lot shorter and also explains what the column does, and sufficiently small that it doesn't even need wrapping.''
> 
> @SnIPeRSnIPeR, would that be a solution for you?

I think we should use "№", "№ п/п" or "#" in Russian, because it's shorter than "Позиция в очереди".

**Comment by SnIPeRSnIPeR** [comment]
Can anyone explain what's difference?

Room wall:



Room wall (personal message set):

**Comment by slook** [comment]
https://hosted.weblate.org/translate/nicotine-plus/nicotine-plus/en/?checksum=a1c925acb1995d5a :

> Mehavoid > "SoulseekQt this string has a translation as "Place"."

This is worth considering? I am not in favour of symbolic "№", "№ п/п" or "#" because it is ambiguous with list index order.

> Can anyone explain what's the difference?

"(personal message set)" appears if you have entered a message into the Room Wall.

**Comment by hboetes** [comment]
@SnIPeRSnIPeR do you understand my point of view? Do you understand working with software requires communication and consideration, learning, trying to understand and patience? Do you understand that simply repeating your own point of view won't help?

Can you imagine Russian people who are new to the program, don't know what that column does and looking at it and seeing `#` don't understand what is going on?

You don't translate for yourself, you translate it for everyone.

**Comment by SnIPeRSnIPeR** [comment]
> "(personal message set)" appears if you have entered a message into the Room Wall... ie; you have activated the feature.

Thanks, will try to translate it properly...

> @SnIPeRSnIPeR do you understand my point of view? Do you understand working with software requires communication and consideration, learning, trying to understand and patience? Do you understand that simply repeating your own point of view won't help?
> 
> Can you imagine Russian people who are new to the program, don't know what that column does and looking at it and seeing `#` don't understand what is going on?
> 
> You don't translate for yourself, you translate it for everyone.

Yes, but I just have wrote my point of view in my messages... Don't think I'm rude or something like that! Getting back to the subject of the conversation... We can change "Позиция в очереди" to "Позиция". It's shortest record in Russian, if anyone don't like "№", "№ п/п" or "#".

**Comment by slook** [comment]
> It is possible to add pop-up hint to "Queue Position" option like at "Room Wall" icon in "Rooms" section?

Unfortunately our GtkTreeView column headers cannot have tooltips at the moment, this is why we can only use the full word description for the label until this is fixed.

> We can change "Позиция в очереди" to "Позиция". It's shortest record in Russian

It seems like this would be a good compromise until a better solution can be found and implemented cleanly.

Thank you for taking the time to properly test your translations in context of the effect this has on the UI/UX, your attention to detail is notable, and this will eventually improve things for everybody in the long run.

**Comment by hboetes** [comment]
@SnIPeRSnIPeR 
> We can change "Позиция в очереди" to "Позиция"

Yes, let's use "Позиция", that's perfectly clear. I wish it were possible that the text changes to "#" as soon as the column gets too small to hold the text. But that is not the case yet.

I'll prepare a pull request that will change the English version to "queue", so all other translation will be shorter as well, eventually. Of course this PR will only be added, after 3.2.1 is released.

File: po/ru.po [hunk 1]
```diff
@@ -1316,7 +1316,7 @@ msgstr "--- старые сообщения выше ---"
 
 #: pynicotine/gtkgui/chatrooms.py:670 pynicotine/gtkgui/chatrooms.py:671
 msgid "Room wall (personal message set)"
-msgstr "Стена комнаты (набор личных сообщений)"
```
Comment (mathiascode) [inline po/ru.po:1319]: @SnIPeRSnIPeR Why this change?

**Comment by mathiascode** [comment]
> "(personal message set)" appears if you have entered a message into the Room Wall... ie; you have activated the feature.

On second thought, it's probably not needed anymore. Your personal message now appears in the text entry, and it's faster to open the room wall than waiting for the tooltip to appear.

**Comment by SnIPeRSnIPeR** [comment]
> Unfortunately our GtkTreeView column headers cannot have tooltips at the moment, this is why we can only use the full word description for the label until this is fixed.

Ok.

> It seems like this would be a good compromise until a better solution can be found and implemented cleanly.
> 
> Thank you for taking the time to properly test your translations in context of the effect this has on the UI/UX, your attention to detail is notable, and this will eventually improve things for everybody in the long run.

Yep, will see.

> Yes, let's use "Позиция", that's perfectly clear. I wish it were possible that the text changes to "#" as soon as the column gets too small to hold the text. But that is not the case yet.
> 
> I'll prepare a pull request that will change the English version to "queue", so all other translation will be shorter as well, eventually. Of course this PR will only be added, after 3.2.1 is released.

It would be nice, but you need to add some new code :) There is one more good variant for "Queue Position": "Очередь".

Ok, thanks.

> On second thought, it's probably not needed anymore. Your personal message now appears in the text entry, and it's faster to open the room wall than waiting for the tooltip to appear.

Exactly! That's why I edited it to "Room wall" in Russian.

**Comment by hboetes** [comment]
@SnIPeRSnIPeR 
We really like suggestions. If a string seems odd or can be improved, please create an issue, and we can think about it and discuss it.

If you simply edit a string to your liking, we may never notice and then a good idea is lost, or you may create confusion because of a problem you have never thought about.

So please, translate all strings as literal as possible, and discus your thoughts in an issue.

Thanks for your hard work.

**Comment by SnIPeRSnIPeR** [comment]
> We really like suggestions. If a string seems odd or can be improved, please create an issue, and we can think about it and discuss it.
> 
> If you simply edit a string to your liking, we may never notice and then a good idea is lost, or you may create confusion because of a problem you have never thought about.
> 
> So please, translate all strings as literal as possible, and discus your thoughts in an issue.
> 
> Thanks for your hard work.

Ok, I got you! Thanks :)

I just have translated new strings like "Open in Folder" in "Uploads" and "Downloads" section.

**Comment by hboetes** [comment]
Awesome, much appreciated!
